### ISSUE TITLE
Invalid argument supplied for foreach()

### ISSUE THREAD

**Issue by LostInInaka** [kalebheitzman/grav-plugin-events#25]
v1.0.13 crashes on front page load 

```
Whoops\Exception\ErrorException thrown with message "Invalid argument supplied for foreach()"

Stacktrace:
#16 Whoops\Exception\ErrorException in /var/www/public_html/classes/thesis/user/plugins/events/classes/events.php:297
#15 Whoops\Run:handleError in /var/www/public_html/classes/thesis/user/plugins/events/classes/events.php:297
#14 Events\Events:initEventStack in /var/www/public_html/classes/thesis/user/plugins/events/classes/events.php:216
#13 Events\Events:instances in /var/www/public_html/classes/thesis/user/plugins/events/events.php:196
#12 Grav\Plugin\EventsPlugin:onPagesInitialized in /var/www/public_html/classes/thesis/vendor/symfony/event-dispatcher/EventDispatcher.php:184
#11 call_user_func in /var/www/public_html/classes/thesis/vendor/symfony/event-dispatcher/EventDispatcher.php:184
#10 Symfony\Component\EventDispatcher\EventDispatcher:doDispatch in /var/www/public_html/classes/thesis/vendor/symfony/event-dispatcher/EventDispatcher.php:46
#9 Symfony\Component\EventDispatcher\EventDispatcher:dispatch in /var/www/public_html/classes/thesis/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php:23
#8 RocketTheme\Toolbox\Event\EventDispatcher:dispatch in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:310
#7 Grav\Common\Grav:fireEvent in /var/www/public_html/classes/thesis/system/src/Grav/Common/Processors/PagesProcessor.php:18
#6 Grav\Common\Processors\PagesProcessor:process in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:121
#5 Grav\Common\Grav:Grav\Common\{closure} in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:396
#4 Grav\Common\Grav:Grav\Common\{closure} in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:372
#3 call_user_func_array in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:372
#2 Grav\Common\Grav:__call in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:122
#1 Grav\Common\Grav:measureTime in /var/www/public_html/classes/thesis/system/src/Grav/Common/Grav.php:122
#0 Grav\Common\Grav:process in /var/www/public_html/classes/thesis/index.php:45

```

v1.0.12 was working find, and everything else works once the plugin is removed.

Any ideas?

**Comment by kalebheitzman** [comment]
Do you have any events defined?

**Comment by LostInInaka** [comment]
I do not, was planning on doing that today - might that be the possible reason why it is faulting out?

**edit:** tried adding a simple event in one of the file - and produces the same error 
**edit 2:** rolled back to 1.0.12 and no problems - loads just fine

**Comment by kalebheitzman** [comment]
It most likely is. I'll code in a hot fix for this. I guess I'm not checking to see if events have been defined or not. Oversight on my part!

**Comment by kalebheitzman** [comment]
I've been unable to reproduce this error. If you could zip your events directory up and let me look at it I might be able to figure out what's going on.

**Comment by kalebheitzman** [comment]
Can you try adding the following code to classes/events.php around line 230? I think there's a bigger problem but I have no idea what it is. The if statement checks to see if events exist and then if they don't returns a blank array to the next process that needs it.

```
	private function initEventStack()
	{
		if ( count( $this->eventPages ) )
			return [];

		foreach ($this->eventPages as $eventPage) {
			$event = [];
```

**Comment by Xarcell** [comment]
I can confirm this error. This happens on a fresh install through softaculous, email plugin update, install date plugin, install event plugin, then crash.

A Test Site which may or may not be around when you check it: http://test-apps.xarcell-studios.com/grav/
PHP Info for the site: http://test-apps.xarcell-studios.com/php-info.php

**Comment by LostInInaka** [comment]
@kalebheitzman I can, but it was a fresh download with no coding mods - but if you are still interested I am more than happy to zip it up for you

**Comment by kalebheitzman** [comment]
Fixed this with Release 1.0.14.
